js/coinsbit.js

```diff
@@ -0,0 +1,520 @@
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicGetDepthResult (this.extend (request, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks');
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        // 'since' param of the request is required a tid as a value.
+        // The exchange will return the trades, starting with this trade id
+        if ('tid' in params) {
+            request['since'] = params.tid;
+        } else {
+            request['since'] = 0;
+        }
+        const trades = await this.publicGetHistoryResult (this.extend (request, params));
+        return this.parseTrades (trades, market, since, limit, params);
+    }
+
+    parseTrade (trade, market) {
+        const symbol = market['symbol'];
+        const id = this.safeString (trade, 'tid');
+        let timestamp = undefined;
+        if ('date' in trade) {
+            timestamp = this.safeTimestamp (trade, 'date');
+        } else {
+            timestamp = this.safeTimestamp (trade, 'time');
+        }
+        const datetime = this.iso8601 (timestamp);
+        const side = this.safeString (trade, 'type');
+        const price = this.safeFloat (trade, 'price');
+        const amount = this.safeFloat (trade, 'amount');
+        const cost = this.safeFloat (trade, 'total');
+        return {
+            'info': trade,
+            'id': id,
+            'timestamp': timestamp,
+            'datetime': datetime,
+            'symbol': symbol,
+            'order': undefined,
+            'type': undefined,
+            'side': side,
+            'takerOrMaker': undefined,
+            'price': price,
+            'amount': amount,
+            'cost': cost,
+            'fee': undefined,
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostAccountBalances (params);
+        const balances = this.safeValue (response, 'result');
+        const currencies = Object.keys (balances);
+        const parsedBalances = {};
+        const free = {};
+        const used = {};
+        const total = {};
+        for (let currencyIndex = 0; currencyIndex < currencies.length; currencyIndex++) {
+            const currency = currencies[currencyIndex];
+            const balance = balances[currency];
+            const parsedBalance = {
+                'free': this.safeFloat (balance, 'available'),
+                'used': this.safeFloat (balance, 'freeze'),
+                'total': this.safeFloat (balance, 'available') + this.safeFloat (balance, 'freeze'),
```

Comment:
https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

Reply:
We don't need to calculate the `total` manually, because that is handled in parseBalance, we should rather call `return this.parseBalance (result)` at the end of this method.